tests/util/test_logging.py

```diff
@@ -49,6 +52,50 @@ async def test_logging_with_queue_handler() -> None:
     assert simple_queue.empty()
 
 
+@pytest.mark.parametrize(
+    ("version", "exc", "exc_info"),
+    [
+        ("2023.10.0", HomeAssistantError, None),
+        ("2023.10.0b0", HomeAssistantError, tuple),
+        ("2023.10.0", ConfigError, None),
+        ("2023.10.0b0", ConfigError, tuple),
+        ("2023.10.0", KeyError, tuple),
+        ("2023.10.0b0", KeyError, tuple),
+    ],
+)
+async def test_suppressed_logging_stack_trace(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+    version: str,
+    exc: Exception,
+    exc_info: type[tuple] | None,
+) -> None:
+    """Test logging stack trace on stable builds.
+
+    Stack traces for HomeAssistantError should be suppressed for stable builds.
+    """
+    logger = logging.getLogger("")
+    with patch("homeassistant.core.__version__", version):
+        log_filter = logging_util.SuppressHomeAssistantErrorStackTrace()
+        logger.addFilter(log_filter)
+
+    try:
+        raise exc("Test exception")
+    except exc as ex:
+        logger.exception(ex)
+
+    logger.removeFilter(log_filter)
+
+    assert len(caplog.records) == 1
+    record = caplog.records[0]
+    assert "Test exception" in record.message
+    assert (
+        record.exc_info is exc_info
+        or isinstance(record.exc_info, tuple)
```

Comment:
Can we avoid the `or` here? Each case should only have one way to assert.

Reply:
Eg can we pass the exception info instance as a parameter and just assert with that?